test/00/t0033a.sh

```diff
@@ -48,7 +48,7 @@ proc afterMinskyStarted {} {uplevel #0 {
   }
  minsky.copy
  saveSelectionAsFile xxx.mky
- group.get [paste]
+ group.get [pasteGroup]
```

Comment:
pasteItems should leave all the items in the selection, so if followed by group, we should end up in the same state as tested by these tests.

Reply:
Okay, I've tried this. When I use PasteItems followed by CreateGroup in t0036a.sh, I get the error:

assertion: [model.numGroups]==2 failed: 
type eval line 11 cmd {assert {[model.numGroups]==2}} proc ::afterMinskyStarted
FAILED

Reply:
You can add a "puts [model.numGroups]" to see what minsky thinks it is.

Reply:
Thanks, now It finds the correct number of groups, but there is still an problem in these two lines:

  # retrieve inner group
  findObject Group
  \$item.groups.@elem 0

So, when an inner group has to be fetched. The error is:

2   (THIS 2 IS RETURNED BY puts [model.numGroups])
index 0 out of boundsindex 0 out of bounds
index 0 out of bounds
    while executing
"$item.groups.@elem 0"
    ("uplevel" body line 16)
    invoked from within
"uplevel #0 {
  openNamedFile /home/wd/minsky_dev/examples/GoodwinLinear02.mky
  assert {[findObject Group]}
  set item minsky.canvas.item
  canvas.mou..."
    (procedure "afterMinskyStarted" line 1)
    invoked from within
"afterMinskyStarted"
    invoked from within
"if {[llength [info commands afterMinskyStarted]]>0} {
    afterMinskyStarted
}"
    (file "/home/wd/minsky_dev/gui-tk/minsky.tcl" line 1649)
FAILED

Reply:
This is saying that the first group found has no groups contained in it.

I'd suggest adding a minsky.save call to save the state of the model to a file, then opening the save file in Minsky to get a picture of what's going on.

Reply:
NB - note the tests work within a temporary directory, which gets deleted once the test is done, so save to an absolute known path. Otherwise do "here=`pwd` in the top level Minsky directory, then copy past the cat command that creates the input file, and run that directly:
gui-tk/minsky input.tcl

Reply:
Thanks. I've done what you suggested in your first message. When I perform pasteItems with the changes I made to paste loose items as ungrouped on a new Canvas, not only the original "Philips Curve" group appears, but also the small phantom group that Hal also picked up in a recent ticket. I think the script then fails when it tries to retrieve the contents of the phantom group. I attach a screen cap of what is saved when I use pasteItems instead of pasteGroup (which is just the original minsky.paste()).
![t0036a sh_screencap](https://user-images.githubusercontent.com/46541934/71346411-1e18f500-2568-11ea-8a4c-d0ded5c21216.png)

Reply:
Since we don't want to copy/paste groups within groups anymore, i.e., an existing group should copied/pasted as the same group and not a group within a group, and ditto for items, I've modified t0036a.sh to take this into account:

##### insert ecolab script code here
##### use \$ in place of $ to refer to variable contents
##### exit 0 to indicate pass, and exit 1 to indicate failure
##### test copy/pasting of groups
cat >input.tcl <<EOF
source $here/test/assert.tcl
proc afterMinskyStarted {} {uplevel #0 {
  openNamedFile $here/examples/GoodwinLinear02.mky
  assert {[findObject Group]}
  set item minsky.canvas.item
  canvas.mouseDown [expr [\$item.x]-0.55*[\$item.width]] [expr [\$item.y]+0.55*[\$item.height]]
  canvas.mouseUp [expr [\$item.x]+0.55*[\$item.width]] [expr [\$item.y]-0.55*[\$item.height]] 
  assert {[canvas.selection.groups.size]==1}
  newSystem
  paste
  **###### For ticket 1080. There is no longer an outer group when existing groups or items are pasted between canvasses or in the same canvas
  assert {[model.numGroups]==1}
  assert {[model.numItems]==8}
  assert {[model.numWires]==8}**
  tcl_exit
}}
EOF